### DISCUSSION TITLE
Realtime+Storage - subscribe to changes to Objects / Buckets

### DISCUSSION THREAD

**Discussion by skwaugh** [supabase/supabase#2794] [Questions] [answered]
Is it possible to subscribe to changes in `storage.objects`?

The `storage` schema is not exposed within the Supabase UI, and attempting to subscribe via SQL is not possible, either, due to ownership limitations.

`alter publication supabase_realtime add table storage.objects;`
`Query 1 ERROR: ERROR:  must be owner of table objects`

Can I subscribe as `supabase_storage_admin`?  How?

**Comment by skwaugh** [comment, marked as answer]
Nevermind, all good.  Resolved.

For anyone having the same permissions issues -

I was executing the command via a 3rd-party Postgres GUI client, and was logged in as the user `postgres`.
Instead, run `alter publication supabase_realtime add table storage.objects;` from the SQL console on supabase.io where the default user is `supabase_admin`.

You can determine your current user by executing `select current_user;`
`select current_schemas(False);` displays the applicable paths for `current_user`

**Comment by jpohhhh** [comment]
I owe you, big time. This is the only resource I found after an hour of Googling, and I was about to give up and start duplicating data in my own table. Had written my whole sync architecture around that I'd be able to do this, so that would have been Very Bad and wasted a ton of time. Cheers.

EDIT: to introduce some search times, supabase realtime file storage, supabase channel for files storage updates changes and events.
